core/Cookie.php

```diff
@@ -456,7 +456,7 @@ private static function getSameSiteValueForBrowser($default)
             } else {
                 $userAgent = Http::getUserAgent();
                 $ddFactory = StaticContainer::get(\Piwik\DeviceDetector\DeviceDetectorFactory::class);
-                $deviceDetector = $ddFactory->makeInstance($userAgent);
+                $deviceDetector = $ddFactory->makeInstance($userAgent, Http::getClientHintsFromServerVariables());
```

Comment:
public static function getClientHintsFromServerVariables()
  it seems this method does not skip X_HTTP_REQUESTED_WITH, which is used to identify anonymous browsers